docs/serving/configuration/config-defaults.md

```diff
@@ -30,319 +32,497 @@ data:
 {% endraw %}
 
-**Key**: `revision-ephemeral-storage-limit`
-
-**Default**: `"750M"` (750 megabytes of storage)
+* **Global key:** `revision-ephemeral-storage-limit`
+* **Per-revision annotation key:** `serving.knative.dev/revision-ephemeral-storage-limit`
+* **Possible values:** integer
+* **Default:** `"750M"` (750 megabytes of storage)
 
 **Example:**
-```yaml
-apiVersion:  v1
-kind:  ConfigMap
-metadata:
-  name:  config-defaults
-  namespace:  knative-serving
-data:
-  revision-ephemeral-storage-limit: "750M" # 750 megabytes of storage
-```
 
-### Container Name Template
+=== "Global (ConfigMap)"
+    ```yaml
+    apiVersion:  v1
+    kind:  ConfigMap
+    metadata:
+      name:  config-defaults
+      namespace:  knative-serving
+    data:
+      revision-ephemeral-storage-limit: "750M"
+    ```
+
+=== "Per Revision"
+    ```yaml
+    apiVersion: serving.knative.dev/v1
+    kind: Service
+    metadata:
+      name: helloworld-go
+      namespace: default
+    spec:
+      template:
+        metadata:
+          annotations:
+            serving.knative.dev/revision-ephemeral-storage-limit: "750M"
+        spec:
+          containers:
+            - image: gcr.io/knative-samples/helloworld-go
+    ```
+
+### Container name template
 {% raw %}
-container-name-template contains a template for the default
-container name, if none is specified.  This field supports
-Go templating and is supplied with the ObjectMeta of the
-enclosing Service or Configuration, so values such as
-{{.Name}} are also valid.
+The `container-name-template` value provides a template for the default
+container name if no container name is specified. This field supports Go templating and is supplied by the `ObjectMeta` of the enclosing Service or Configuration, so values such as `{{.Name}}` are also valid.
 {% endraw %}
 
-**Key**: `container-name-template`
-
-**Default**: `"user-container"`
+* **Global key:** `container-name-template`
+* **Per-revision annotation key:** `serving.knative.dev/container-name-template`
+* **Possible values:** string
+* **Default:** `"user-container"`
 
 **Example:**
-```yaml
-apiVersion:  v1
-kind:  ConfigMap
-metadata:
-  name:  config-defaults
-  namespace:  knative-serving
-data:
-  container-name-template: "user-container"
-```
 
-### Container Concurrency
+=== "Global (ConfigMap)"
+    ```yaml
+    apiVersion:  v1
+    kind:  ConfigMap
+    metadata:
+      name:  config-defaults
+      namespace:  knative-serving
+    data:
+      container-name-template: "user-container"
+    ```
+
+=== "Per Revision"
+    ```yaml
+    apiVersion: serving.knative.dev/v1
+    kind: Service
+    metadata:
+      name: helloworld-go
+      namespace: default
+    spec:
+      template:
+        metadata:
+          annotations:
+            serving.knative.dev/container-name-template: "user-container"
```

Comment:
This annotation doesn't exist - users can set the `Spec.Containers[*].Name` attribute on the `Revision` explicitly

Reply:
see previous comments

Reply:
You can set the name of the container like this

```yaml
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  name: helloworld-go
  namespace: default
spec:
  template:
    spec:
      containers:
        - name: some-name
          image: gcr.io/knative-samples/helloworld-go
```